thirdeye/thirdeye-frontend/app/pods/self-serve/create-alert/controller.js

```diff
@@ -256,37 +258,70 @@ export default Ember.Controller.extend({
+   * @param {String} message - the appropriate error message
+   * @return {undefined}
+   */
+  setErrorState(error, message) {
+    this.setProperties({
+      isReplayError: true,
+      failureMessage: `${message}. (${error})`
+    });
+    throw error;
+  },
+
+  /**
    * Sends a request to begin advanced replay for a metric. The replay will fetch new
    * time-series data based on user-selected sensitivity settings.
    * @method triggerReplay
-   * @param {Object} functionObj - the alert object
-   * @param {Object} groupObj - the alert notification group object
    * @param {Number} newFuncId - the id for the newly created function (alert)
    * @return {Ember.RSVP.Promise}
    */
-  triggerReplay(functionObj, groupObj, newFuncId) {
-    const startTime = moment().subtract(1, 'month').endOf('day').format("YYYY-MM-DD");
-    const startStamp = moment().subtract(1, 'day').endOf('day').valueOf();
-    const endTime = moment().subtract(1, 'day').endOf('day').format("YYYY-MM-DD");
-    const endStamp = moment().subtract(1, 'month').endOf('day').valueOf();
-    const granularity = this.get('graphConfig.granularity').toLowerCase();
-    const postProps = {
-      method: 'POST',
-      timeout: 120000, // 2 min
-      headers: { 'Content-Type': 'Application/Json' }
-    };
-    const replayApi = {
-      minute: `/replay/singlefunction?functionId=${newFuncId}&start=${startTime}&end=${endTime}`,
-      hour: `/${newFuncId}/replay?start=${startTime}&end=${endTime}`,
-      day: `/replay/function/${newFuncId}?start=${startTime}&end=${endTime}&goal=1.0&evalMethod=F1_SCORE&includeOriginal=false&tune=\{"pValueThreshold":\[0.001,0.005,0.01,0.05\]\}`,
-      reports: `/thirdeye/email/generate/metrics/${startStamp}/${endStamp}?metrics=${functionObj.metric}&subject=Your%20Metric%20Has%20Onboarded%20To%20Thirdeye&from=thirdeye-noreply@linkedin.com&to=${groupObj.recipients}&smtpHost=email.corp.linkedin.com&smtpPort=25&includeSentAnomaliesOnly=true&isApplyFilter=true`
-    };
-    const allowedGranularity = ['minute', 'hour', 'day'];
-    const gkey = allowedGranularity.includes(granularity) ? granularity : 'minute';
-
-    return new Ember.RSVP.Promise((resolve) => {
-      fetch('/detection-job' + replayApi[gkey], postProps).then((res) => checkStatus(res, 'post'));
-    });
+  triggerReplay(newFuncId) {
+    const startTime = moment().subtract(1, 'month').endOf('day').utc().format("YYYY-MM-DDTHH:mm:ss.SSS[Z]");
+    const endTime = moment().subtract(1, 'day').endOf('day').utc().format("YYYY-MM-DDTHH:mm:ss.SSS[Z]");
+
+    this.callCloneAlert(newFuncId)
+      .then((clonedId) => {
+        this.set('isReplayTriggeredSuccess', true);
+        return this.callReplayStart(clonedId, startTime, endTime);
+      })
+      .catch((error) => {
+        this.setErrorState(error, 'Failed to clone new alert function');
+      })
+      .then((jobId) => {
+        this.set('isReplayStatusPending', false);
```

Comment:
this `.then` would only be called if an error was caught on line 315. Is this the intended behavior?